### PR TITLE
Changing updating deleting

### PR DESCRIPTION
**Pull request by AntonBakum**

### CODE REVIEW

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx [hunk 1]
```diff
@@ -7,73 +7,86 @@ import {
...
+                    <Button
+                      sx={styles.closeButton}
+                      onClick={props.onToggleCreateStatusClick}
+                    >
+                      Close
+                    </Button>
+                  </div>
+                </Grid>
+                <Grid item xs={6}>
+                  <div style={styles.buttonsWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: Usually, it is named formActions , blablablaActions... (optional to rename)

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    <Dialog open={props.deleteModalOpen}>
+      <div style = {styles.dialogTitleWrapper}>
+        <DialogTitle style={styles.dialogTitle}>
+          Click on the confirmation button to delete the category
+        </DialogTitle>
+      </div>
+      <DialogContent>
+        <Grid container>
+          <Grid item xs={6}>
+            <div style={styles.buttonsWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: A lot of containers for buttons. simplify if it is possible.

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+const DeleteCategoryComponent = (props: Props) => {
+  return (
+    <Dialog open={props.deleteModalOpen}>
+      <div style = {styles.dialogTitleWrapper}>
+        <DialogTitle style={styles.dialogTitle}>
+          Click on the confirmation button to delete the category
+        </DialogTitle>
+      </div>
+      <DialogContent>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx:24]: It is not content. You should use  <DialogActions> for buttons https://mui.com/material-ui/react-dialog/#ResponsiveDialog.tsx

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  onToggleDeleteStatusClick: () => void;
+  onSubmit: () => void;
+}
+
+const DeleteCategoryComponent = (props: Props) => {
+  return (
+    <Dialog open={props.deleteModalOpen}>
+      <div style = {styles.dialogTitleWrapper}>
+        <DialogTitle style={styles.dialogTitle}>
+          Click on the confirmation button to delete the category
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: it should be
 ```
<DialogContent>
          <DialogContentText>
            Click on the confirmation button to delete the category
          </DialogContentText>
        </DialogContent>
```

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  deleteModalOpen: boolean;
+  onToggleDeleteStatusClick: () => void;
+  onSubmit: () => void;
+}
+
+const DeleteCategoryComponent = (props: Props) => {
+  return (
+    <Dialog open={props.deleteModalOpen}>
+      <div style = {styles.dialogTitleWrapper}>
+        <DialogTitle style={styles.dialogTitle}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: DialogTitle should be "Delete Category"

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+interface Props {
+  deleteModalOpen: boolean;
+  onToggleDeleteStatusClick: () => void;
+  onSubmit: () => void;
+}
+
+const DeleteCategoryComponent = (props: Props) => {
+  return (
+    <Dialog open={props.deleteModalOpen}>
+      <div style = {styles.dialogTitleWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: A lot of wrappers for one component.

File: src/pages/CategoriesPage/Table/DeleteCategoryColumn/DeleteCategoryColumnComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import { styles } from "../../styles";
+
+interface Props {
+    id: number
+    onToggleDeleteStatusClick: () => void
+}
+
+const DeleteCategoryColumnComponent = (props: Props) => {
+    return (
+        <div>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/Table/DeleteCategoryColumn/DeleteCategoryColumnComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/Table/CategoriesTableComponent.tsx [hunk 8]
```diff
@@ -1,17 +1,17 @@
...
 import { styles } from '../styles';
 
 interface Props {
   categories: CategoryModel[];
+  categoryTableColumns: GridColDef[]
 }
 
 const CategoriesTableComponent = (props: Props) => {
   return (
     <div style={styles.tableWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/Table/CategoriesTableComponent.tsx:13]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx (hunk 2 above)
Comment (ValentynCGL) [inline src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx [hunk 9]
```diff
@@ -7,73 +7,86 @@ import {
...
+                  error={
+                    formProps.touched.description &&
+                    Boolean(formProps.errors.description)
+                  }
+                  helperText={
+                    formProps.touched.description &&
+                    formProps.errors.description
+                  }
+                />
+              </div>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx [hunk 10]
```diff
@@ -7,73 +7,86 @@ import {
...
+            description: '',
+          }}
+          validationSchema={validationSchema}
+          onSubmit={(values: CreateUpdateCategoryModel) => {
+            props.onSubmit(values);
+          }}
+        >
+          {(formProps) => (
+            <form onSubmit={formProps.handleSubmit}>
+              <div style={styles.fieldsWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx [hunk 11]
```diff
@@ -7,73 +7,86 @@ import {
...
-  onSubmit: (values: CreateCategoryModel) => void;
+  onSubmit: (values: CreateUpdateCategoryModel) => void;
   onToggleCreateStatusClick: () => void;
 }
 
 const AddNewCategoryComponent = (props: Props) => {
   return (
     <Dialog open={props.createModalOpen}>
-      <DialogTitle sx={styles.dialogTitle}>CREATE CATEGORY</DialogTitle>
+      <div style={styles.dialogTitleWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+            description: props.description,
+          }}
+          validationSchema={validationSchema}
+          onSubmit={(values: CreateUpdateCategoryModel) => {
+            props.onSubmit(values);
+          }}
+        >
+          {(formProps) => (
+            <form onSubmit={formProps.handleSubmit}>
+              <div style={styles.fieldsWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+  onSubmit: (values: CreateUpdateCategoryModel) => void;
+  name: string;
+  description: string;
+}
+
+const UpdateCategoryComponent = (props: Props) => {
+  return (
+    <Dialog open={props.updateModalOpen}>
+      <div style={styles.dialogTitleWrapper}>
+        <DialogTitle sx={styles.dialogTitle}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: Dialog should contain the right structure https://mui.com/material-ui/react-dialog/#ResponsiveDialog.tsx

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+  onToggleUpdateStatusClick: () => void;
+  onSubmit: (values: CreateUpdateCategoryModel) => void;
+  name: string;
+  description: string;
+}
+
+const UpdateCategoryComponent = (props: Props) => {
+  return (
+    <Dialog open={props.updateModalOpen}>
+      <div style={styles.dialogTitleWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+                  variant="outlined"
+                  onChange={formProps.handleChange}
+                  value={formProps.values.name}
+                  error={
+                    formProps.touched.name && Boolean(formProps.errors.name)
+                  }
+                  helperText={formProps.touched.name && formProps.errors.name}
+                />
+              </div>
+              <div style={styles.fieldsWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+                  }
+                  helperText={
+                    formProps.touched.description &&
+                    formProps.errors.description
+                  }
+                />
+              </div>
+              <Grid container>
+                <Grid item xs={6}>
+                  <div style={styles.buttonsWrapper}>
```
Comment (ValentynCGL) [inline src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: Do not use DIV. Use Box instead.

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx (hunk 11 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx (hunk 10 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx (hunk 9 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx (hunk 6 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: Amount of wrappers in this component has been decreased

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx (hunk 5 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: Name of the DialogTitle has been changed

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx (hunk 4 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: I added <DialogContentText> component with the  action subscription to all the dialog windows

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx (hunk 3 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx:24]: New location for buttons is <DialogActions> component

File: src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx (hunk 2 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: The layout and styles have been changed and wrapper divs have been deleted from the source code.

Comment (AntonBakum) [reply src/pages/CategoriesPage/DeleteCategory/DeleteCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/Table/CategoriesTableComponent.tsx (hunk 8 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/Table/CategoriesTableComponent.tsx:13]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/Table/DeleteCategoryColumn/DeleteCategoryColumnComponent.tsx (hunk 7 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/Table/DeleteCategoryColumn/DeleteCategoryColumnComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx (hunk 14 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx (hunk 12 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx (hunk 1 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/AddCategory/AddNewCategoryComponent.tsx]: formActions in the components that have a forms have been renamed according to  this naming convention

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx (hunk 13 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: Dialog structure has been changed

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx (hunk 15 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

File: src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx (hunk 16 above)
Comment (AntonBakum) [reply src/pages/CategoriesPage/UpdateCategory/UpdateCategoryComponent.tsx]: All divs in the modal windows have been replaced by the Box components

**Review by ValentynCGL — APPROVED** [review]

**Pull request merged**